src/compiler/checker.ts

```diff
@@ -8977,17 +8978,35 @@ namespace ts {
                 }
             }
 
-            function typeRelatedToEachType(source: Type, target: UnionOrIntersectionType, reportErrors: boolean): Ternary {
+            function typeRelatedToEachType(source: Type, target: IntersectionType, reportErrors: boolean): Ternary {
                 let result = Ternary.True;
                 const targetTypes = target.types;
+                const saveDisableWeakTypeErrors = disableWeakTypeErrors;
+                disableWeakTypeErrors = true;
                 for (const targetType of targetTypes) {
                     const related = isRelatedTo(source, targetType, reportErrors);
                     if (!related) {
+                        disableWeakTypeErrors = saveDisableWeakTypeErrors;
                         return Ternary.False;
                     }
                     result &= related;
                 }
-                return result;
+                disableWeakTypeErrors = saveDisableWeakTypeErrors;
+                return reportAssignmentToWeakIntersection(source, target, reportErrors) ? Ternary.False : result;

```

Comment:
Oic. Jsut to make sure I understand this correctly because `reportAaassignmentToWeakIntersection` does the check in line 8999 and 9000 -> checking that all constituents are weak type ? .....would you mind add this comment in? it wasn't obvious jsut from looking at the caller